src/widgets/infinite-hits/__tests__/infinite-hits-integration-test.ts

```diff
@@ -0,0 +1,122 @@
+import algoliasearch from 'algoliasearch/lite';
+import { getByText, waitFor, fireEvent } from '@testing-library/dom';
+
+import instantsearch from '../../../index.es';
+import { infiniteHits, configure } from '../../';
+
+describe('infiniteHits', () => {
+  let search;
+  let searchClient;
+  let container;
+
+  let cachedState: any;
+  let cachedHits: any;
+  let customCache;
+
+  beforeEach(() => {
+    searchClient = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76');
```

Comment:
a real index doesn't seem like that stable of a source for tests. Maybe creating an array with counters and using a fake client could be better?

Reply:
It makes sense.
https://github.com/algolia/instantsearch.js/pull/4413/commits/64abc2c6463037e586a8ad9bf0ac666fd088101d
I think I did it correctly, right?